### PR TITLE
refactor: use strings for GRPC uint64 fields

### PR DESCRIPTION
**Pull request by jamaljsr**
This PR is a big refactoring to use strings to transmit large numbers via GRPC and [big.js](https://github.com/MikeMcl/big.js) for large numeric variables within the application. This resolves issues with JS number overflow. 

Since many of the GRPC field types changed from `number` to `string`, a large portion of the codebase needed to be updated for it to even compile. The commits are broken down based on the section of the code that was updated. Each commit will not compile on its own except for the last one. 

There are no functional changes in the app. To test, just navigate around and perform some Loop and Pool actions to confirm that no errors occur.

### CODE REVIEW

**Review by guggero — APPROVED** [review]
tACK! Tried out most features and couldn't find anything not working. LGTM :100: 

I ran the proto compilation on my machine. But because I have `protoc 3.12.x` installed I got quite a large diff. I think we should probably move over to docker for generating the protos and also possibly update the protoc version being used. Going to open a PR for this.

File: app/scripts/build-protos.js [hunk 1]
```diff
@@ -36,7 +36,9 @@ const protoSources = async () => {
     throw new Error(`go.mod did not match pattern ${POOL_VERSION_PATTERN}`);
   }
 
-  console.log(`Found lnd version ${lndVersion[1]} and loop version ${loopVersion[1]}.`);
+  console.log(
+    `Found:\n LND ${lndVersion[1]}\n Loop ${loopVersion[1]}\n Pool poolVersion[1]`,
```
[the remark this answers is not in the document]
Comment (jamaljsr) [reply app/scripts/build-protos.js]: good catch. thanks!

File: app/scripts/build-protos.js [hunk 2]
```diff
@@ -78,6 +80,26 @@ const download = async () => {
...
+ */
+const sanitize = async () => {
+  const filePaths = Object.keys(filePatches).map(name =>
+    join(appPath, '..', 'proto', `${name}.proto`),
+  );
+  for (path of filePaths) {
+    let content = (await fs.readFile(path)).toString();
+    content = content.replace(/^\s*(repeated)? u?int64 ((?!jstype).)*$/gm, match => {
+      // add the jstype descriptor
+      return /^.*];$/.test(match)
```
[the remark this answers is not in the document]
Comment (jamaljsr) [reply app/scripts/build-protos.js:95]: ty sir 🙏

File: app/src/api/loop.ts [hunk 3]
```diff
@@ -138,15 +138,16 @@ class LoopApi extends BaseApi<LoopEvents> {
...
+
   /**
    * Calculates the max routing fee params for loop out. this mimics the loop cli
    * behavior of using 2% of the amount
    * @param amount the amount of the payment
    */
-  private _calcRoutingFee(amount: number) {
-    const routingFeePct = 2;
+  private _calcRoutingFee(amount: Big): Big {
+    const routingFeePct = 0.02;
```
[the remark this answers is not in the document]
Comment (jamaljsr) [reply app/src/api/loop.ts]: Good point. I renamed it

File: proto/auctioneerrpc/auctioneer.proto [hunk 4]
```diff
@@ -48,7 +48,7 @@ message ReserveAccountRequest {
     /*
     The desired value of the account in satoshis.
     */
-    uint64 account_value = 1;
+    uint64 account_value = 1 [jstype = JS_STRING];
```
[the remark this answers is not in the document]
Comment (jamaljsr) [reply proto/auctioneerrpc/auctioneer.proto:51]: Word. That would have made things a big cleaner.

**Comment by jamaljsr** [comment]
Thanks for the feedback and testing. The suggested changes have been made.

**Pull request merged**